src/Syntax/SqlParser.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Syntax;
+
+use function strlen;
+use function substr;
+
+/**
+ * SQL parser.
+ *
+ * This class provides methods to parse SQL statements and extract placeholders from them.
+ */
+class SqlParser
+{
+    /**
+     * @var int Length of SQL statement.
+     */
+    protected int $length;
+
+    /**
+     * @var int Current position in SQL statement.
+     */
+    protected int $position = 0;
+
+    /**
+     * @param string $sql SQL statement to parse.
+     */
+    public function __construct(protected string $sql)
+    {
+        $this->length = strlen($sql);
+    }
+
+    /**
+     * Returns the next placeholder from the current position in SQL statement.
+     *
+     * @param int|null $position Position of the placeholder in SQL statement.
+     *
+     * @return string|null The next placeholder or null if it is not found.
+     */
+    public function getNextPlaceholder(int|null &$position = null): string|null
+    {
+        $result = null;
+        $length = $this->length - 1;
+
+        while ($this->position < $length) {
+            $pos = $this->position++;
+
+            match ($this->sql[$pos]) {
+                ':' => ($word = $this->parseWord()) === ''
+                    ? $this->skipChars(':')
+                    : $result = ':' . $word,
+                '"', "'" => $this->skipQuotedWithoutEscape($this->sql[$pos]),
+                '-' => $this->sql[$this->position] === '-'
+                    ? ++$this->position && $this->skipToAfterChar("\n")
+                    : null,
+                '/' => $this->sql[$this->position] === '*'
+                    ? ++$this->position && $this->skipToAfterString('*/')
+                    : null,
```

Comment:
May be make parser abstract (all current Yii DB implementations has own parser) and make abstract method, that allow implementations of this abstraction add or replace code for symbols?

Reply:
Thought about this. This is SQL-standart implementation but each DBMS has own quoting features.

Ok, I'll make it abstract.

Reply:
Done